### ISSUE TITLE
how do we edit records with errors?

### ISSUE THREAD

**Issue by teixeirak** [SCBI-ForestGEO/SCBImortality#20]
@jess-shue , the last step to getting this system fully working (hopefully by next week, #18 ) is understanding how to edit records with errors. I know you've figured out how to do that; can you please explain? It would be really helpful to have this before Tuesday, as we're hoping to clean up errors that day.

**Comment by teixeirak** [comment]
(actually, the sooner we know the better, so that we can have some time this afternoon to prepare anything needed and clean up errors that don't require a field fix.)

**Comment by jess-shue** [comment]
@teixeirak Do you mean once data have been submitted and run through Valentine's checks? This is not something I have automated and would usually be an edit to the file - I am not that sophisticated.
If you want to edit the raw data, I believe you can re-deploy the quadrat file from FFF's and edit it within the app - then resubmit and repeat the process. Is that the kind of thing you are hoping for?

**Comment by teixeirak** [comment]
Yes, this is after it runs through the checks.  

Then, for example, it might be necessary to go back and edit one value (e.g., % foliage), and we don't want to create a full new record for the tree.

I think your second option (re-deploy and edit within the FF app) would be easiest, but I'll also leave this for @briannakcalkins or @jenajordan  to comment. Perhaps it could make more sense to just edit the single value in a .csv file? (The latter option would be more complicated on @ValentineHerr 's end)

**Comment by jess-shue** [comment]
@teixeirak @briannakcalkins @jenajordan 
OK, yes you can 'dispatch' the submitted form to a user through the FFF's website:  
1. Open the FFF's account and choose 'Legacy Portal'
2. Go to 'Submissions' and in the 'Filters' section on the right select 'SCBI Mortality'
3. Add a filter - this can be the quadrat name from the drop down list:  Quad/lookuplistpicker_1
4. Apply the filter and you should have a list of the submissions for the quadrat (likely only 1)
5. To the left of the submission there is a 'View' button with an arrow. Click on the arrow and 'Dispatch'
6. Type ecoclimlab@gmail.com in the 'Users' box, then click 'Dispatch'
7. Open the FFF's app on the iPad and open the Inbox to view the dispatched file. From there you can edit and resubmit and the submission should replace the original.

**Comment by teixeirak** [comment]
@jess-shue , I was writing a long message and our messages crossed. 

I went into FastField on my phone and am able to pull up and edit old records. That seems very easy, but it would be good to do a test.

One thing that may be a bit awkward is that it looks like you'd be resubmitting the whole quadrat, which would presumably generate new census dates for all trees and reset the surveyor. @ValentineHerr could adjust her code to deal with that though (if a record didn't change, keep the earlier record; if the person who fixed it is different from the original surveyor, list both).

**Comment by teixeirak** [comment]
It would be good if we could do a practice fix on one that doesn't require a field fix. For example, the records in [`dead_but_crown_thinning_less_than_5.csv`](https://github.com/SCBI-ForestGEO/SCBImortality/blob/main/testthat/reports/requires_field_fix/dead_but_crown_thinning_less_than_5.csv) are all wrong for the same reason: `Crown thinning` is recorded 0 (not a viable code), whereas looking at other fields makes it clear that they should all be `5` (crown completely dead).

**Comment by teixeirak** [comment]
Perhaps both methods work? Is it necessary to go through your steps, Jess? I suppose that your approach will allow us to isolate single records, and thereby avoid the problem I described above? I'm going to try to experiment in a few minutes.

**Comment by teixeirak** [comment]
> @teixeirak @briannakcalkins @jenajordan
> OK, yes you can 'dispatch' the submitted form to a user through the FFF's website:
> 
>     1. Open the FFF's account and choose 'Legacy Portal'
> 
>     2. Go to 'Submissions' and in the 'Filters' section on the right select 'SCBI Mortality'
> 
>     3. Add a filter - this can be the quadrat name from the drop down list:  Quad/lookuplistpicker_1

@jess-shue , I got stuck at this point. I don't see quadrat name as a choice:
<img width="506" alt="image" src="https://user-images.githubusercontent.com/6355854/124300737-deec3c80-db2c-11eb-8b94-9fe77aa49be2.png">

**Comment by teixeirak** [comment]
... but I was able to sort by submission ID, which is given in Valentine's error report files.

**Comment by teixeirak** [comment]
Okay, I got through that process and submitted a correction to the first one in the list:
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/6355854/124301539-fa0b7c00-db2d-11eb-96ea-f07b25555bcd.png">

I'll now download the latest and confirm that that worked.

Future errors of this type should be avoidable: @jess-shue , "0" should not be an option; can you please remove?

**Comment by teixeirak** [comment]
@jess-shue , do you know the difference between following the steps you outlined above (quite clunky) vs just going in and pulling up the records in editing the existing records in the FFF app, editing, and resubmitting?

**Comment by teixeirak** [comment]
The record is now corrected in the excel file, and GitHub actions is finished; however, the tests report has not updated. @ValentineHerr , do you know what's going on there?

**Comment by teixeirak** [comment]
Note: we'll probably need a single list of fixes needed per quadrat in order to make this workable in the field. I'll come back to this after lunch.

**Comment by jess-shue** [comment]
@teixeirak If you open the app and pull up the tree with the error you won't be editing the original submission and will submit a duplicate of that tree. You won't be editing an existing record that was submitted - you'll be starting with the empty data for that tree. The app does not update with the submissions.

**Comment by jess-shue** [comment]
> > @teixeirak @briannakcalkins @jenajordan
> > OK, yes you can 'dispatch' the submitted form to a user through the FFF's website:
> > ```
> > 1. Open the FFF's account and choose 'Legacy Portal'
> > 
> > 2. Go to 'Submissions' and in the 'Filters' section on the right select 'SCBI Mortality'
> > 
> > 3. Add a filter - this can be the quadrat name from the drop down list:  Quad/lookuplistpicker_1
> > ```
> 
> @jess-shue , I got stuck at this point. I don't see quadrat name as a choice:
> <img alt="image" width="506" src="https://user-images.githubusercontent.com/6355854/124300737-deec3c80-db2c-11eb-8b94-9fe77aa49be2.png">

That should be the top results on the list, if you scroll down you will see this:
![image](https://user-images.githubusercontent.com/37407801/124305691-6ccb2600-db33-11eb-89ff-cf19f560cdbd.png)

**Comment by jess-shue** [comment]
> Okay, I got through that process and submitted a correction to the first one in the list:
> <img alt="image" width="1207" src="https://user-images.githubusercontent.com/6355854/124301539-fa0b7c00-db2d-11eb-96ea-f07b25555bcd.png">
> 
> I'll now download the latest and confirm that that worked.
> 
> Future errors of this type should be avoidable: @jess-shue , "0" should not be an option; can you please remove?

@teixeirak Which field is this in? Sorry, but I can't see what this error is for from your comment. The zero's were included based on the 2020 mortality survey values.

**Comment by teixeirak** [comment]
> @teixeirak Crown thinning for ash/Chioanthus, correct? Everyone will need to update the app on the iPads for this change to be seen in the field.

Correct. Sorry I forgot to specify earlier.

**Comment by teixeirak** [comment]
Thanks for all the answers above, @jess-shue . I think it all makes sense. I'm waiting on issue #23 before moving forward. 

@briannakcalkins or @jenajordan , once issue #23 is complete, we'll need to follow Jess's instructions above to "dispatch" all quadrats with errors back to FastField for correction. That's a bit of a clunky process, and I'm not sure we can get it all done by Tuesday.

**Comment by ValentineHerr** [comment]
Hmm,  still getting an error because of this rgdal package, i'll let you know when fixed

**Comment by ValentineHerr** [comment]
(sorry my commit and comment above were supposed to be in another issue - but the test  doesn't break anymore). Working in issue #23 now

**Comment by teixeirak** [comment]
I've created instructions to edit records here: https://github.com/SCBI-ForestGEO/SCBImortality/blob/main/Protocols/editing%20records%20with%20errors.md. 

I'm flagging @briannakcalkins and @jenajordan so you know that these instructions exist.

This issue can now be closed!
